platform/features/layout/src/LayoutController.js

```diff
@@ -156,8 +175,8 @@ define(
             // Watch for changes to the grid size in the model
             $scope.$watch("model.layoutGrid", updateGridSize);
 
-            // Position panes when the model field changes
-            $scope.$watch("model.composition", lookupPanels);
+            // Update composed objects on screen, and position panes
+            $scope.$watch("model.composition", refreshComposition);
```

Comment:
I believe a `$watch` here would fail to detect changes made in-place to `model.composition`. `$watchCollection` would be more appropriate here, although for large compositions it could be costly.

If `$watchCollection` is too heavy, you could use a mutation listener and explicitly compare the `composition` array to see if there are any changes.